### PR TITLE
Velocity nodes

### PR DESCRIPTION
**Pull request by cboesch-gpsw**
This adds the support for ~~motion~~ velocity nodes. The branch also includes many fixes and cleanups to the existing easing/animations.

### CODE REVIEW

File: libnodegl/nodegl.h.in [hunk 1]
```diff
@@ -677,6 +677,21 @@ NGL_API int ngl_easing_evaluate(const char *name, const double *args, int nb_arg
 NGL_API int ngl_easing_solve(const char *name, const double *args, int nb_args,
                              const double *offsets, double v, double *t);
 
+/**
+ * Derivate an easing for a given time t
```
Comment (mbouron-gpsw) [inline libnodegl/nodegl.h.in:687]: Missing period.

Comment (mbouron-gpsw) [reply libnodegl/nodegl.h.in:687]: It's also the case for ngl_easing_solve/evaluate().

File: libnodegl/node_render.c [hunk 2]
```diff
@@ -104,6 +104,10 @@ struct render_priv {
                                           NGL_NODE_STREAMEDVEC4,    \
                                           NGL_NODE_STREAMEDMAT4,    \
                                           NGL_NODE_TIME,            \
+                                          NGL_NODE_VELOCITYFLOAT,   \
+                                          NGL_NODE_VELOCITYVEC2,    \
+                                          NGL_NODE_VELOCITYVEC3,    \
+                                          NGL_NODE_VELOCITYVEC4,    \
```
Comment (mbouron-gpsw) [inline libnodegl/node_render.c:110]: Shouldn't this also be made available to the compute node ?

**Comment by mbouron-gpsw** [comment]
Tested on my machine (Linux, Windows), tests pass. Other than my remarks (and the offline ones), LGTM.

File: libnodegl/nodegl.h.in (hunk 1 above)
Comment (cboesch-gpsw) [reply libnodegl/nodegl.h.in]: > Missing period.

Fixed

> It's also the case for ngl_easing_solve/evaluate().

Added the following commit: `api: add a few missing periods in docstrings`

File: libnodegl/node_render.c (hunk 2 above)
Comment (cboesch-gpsw) [reply libnodegl/node_render.c:110]: Oups. Added.

**Review by mbouron-gpsw — APPROVED** [review]

**Pull request merged**